{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2cd219ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b3565392",
   "metadata": {},
   "source": [
    "### Data DisGeNET"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "430620c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>DSI</th>\n",
       "      <th>DPI</th>\n",
       "      <th>PLI</th>\n",
       "      <th>protein_class_name</th>\n",
       "      <th>protein_class</th>\n",
       "      <th>NofDiseases</th>\n",
       "      <th>NofPmids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.700</td>\n",
       "      <td>0.538</td>\n",
       "      <td>4.991700e-09</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>0.529</td>\n",
       "      <td>0.769</td>\n",
       "      <td>4.522900e-11</td>\n",
       "      <td>Enzyme modulator</td>\n",
       "      <td>DTO_05007584</td>\n",
       "      <td>147</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>A2MP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>NAT1</td>\n",
       "      <td>0.536</td>\n",
       "      <td>0.846</td>\n",
       "      <td>1.929400e-14</td>\n",
       "      <td>Enzyme</td>\n",
       "      <td>DTO_05007624</td>\n",
       "      <td>133</td>\n",
       "      <td>184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>NAT2</td>\n",
       "      <td>0.451</td>\n",
       "      <td>0.885</td>\n",
       "      <td>3.274400e-06</td>\n",
       "      <td>Enzyme</td>\n",
       "      <td>DTO_05007624</td>\n",
       "      <td>311</td>\n",
       "      <td>627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21666</th>\n",
       "      <td>115482722</td>\n",
       "      <td>H3P33</td>\n",
       "      <td>0.736</td>\n",
       "      <td>0.346</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21667</th>\n",
       "      <td>115482723</td>\n",
       "      <td>H3P40</td>\n",
       "      <td>0.506</td>\n",
       "      <td>0.846</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>175</td>\n",
       "      <td>202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21668</th>\n",
       "      <td>115561815</td>\n",
       "      <td>LNC-LBCS</td>\n",
       "      <td>0.890</td>\n",
       "      <td>0.077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21669</th>\n",
       "      <td>115804232</td>\n",
       "      <td>CEROX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21670</th>\n",
       "      <td>115891964</td>\n",
       "      <td>MIR223HG</td>\n",
       "      <td>0.861</td>\n",
       "      <td>0.077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21671 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          geneId geneSymbol    DSI    DPI           PLI protein_class_name  \\\n",
       "0              1       A1BG  0.700  0.538  4.991700e-09           Receptor   \n",
       "1              2        A2M  0.529  0.769  4.522900e-11   Enzyme modulator   \n",
       "2              3      A2MP1    NaN    NaN           NaN                NaN   \n",
       "3              9       NAT1  0.536  0.846  1.929400e-14             Enzyme   \n",
       "4             10       NAT2  0.451  0.885  3.274400e-06             Enzyme   \n",
       "...          ...        ...    ...    ...           ...                ...   \n",
       "21666  115482722      H3P33  0.736  0.346           NaN                NaN   \n",
       "21667  115482723      H3P40  0.506  0.846           NaN                NaN   \n",
       "21668  115561815   LNC-LBCS  0.890  0.077           NaN                NaN   \n",
       "21669  115804232     CEROX1    NaN    NaN           NaN                NaN   \n",
       "21670  115891964   MIR223HG  0.861  0.077           NaN                NaN   \n",
       "\n",
       "      protein_class  NofDiseases  NofPmids  \n",
       "0      DTO_05007575           27        20  \n",
       "1      DTO_05007584          147       145  \n",
       "2               NaN            1         1  \n",
       "3      DTO_05007624          133       184  \n",
       "4      DTO_05007624          311       627  \n",
       "...             ...          ...       ...  \n",
       "21666           NaN           19        15  \n",
       "21667           NaN          175       202  \n",
       "21668           NaN            3         1  \n",
       "21669           NaN            1         1  \n",
       "21670           NaN            4         1  \n",
       "\n",
       "[21671 rows x 9 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Loading the CSV file into a DataFrame\n",
    "gene_associations = pd.read_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Data\\Protein\\Disgenet\\gene_associations.tsv', delimiter='\\t')\n",
    "gene_associations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "95ea2ef9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snpId</th>\n",
       "      <th>geneId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>sourceId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs7980288</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs7980288</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs3832852</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs669</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs669</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361962</th>\n",
       "      <td>rs556554798</td>\n",
       "      <td>115072896</td>\n",
       "      <td>SLCO1B3-SLCO1B7</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361963</th>\n",
       "      <td>rs11045626</td>\n",
       "      <td>115072896</td>\n",
       "      <td>SLCO1B3-SLCO1B7</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361964</th>\n",
       "      <td>rs4149117</td>\n",
       "      <td>115072896</td>\n",
       "      <td>SLCO1B3-SLCO1B7</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361965</th>\n",
       "      <td>rs629444</td>\n",
       "      <td>115482690</td>\n",
       "      <td>H2BP5</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361966</th>\n",
       "      <td>rs3803700</td>\n",
       "      <td>115804232</td>\n",
       "      <td>CEROX1</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>361967 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              snpId     geneId       geneSymbol sourceId\n",
       "0         rs7980288          2              A2M      VEP\n",
       "1         rs7980288          2              A2M    DBSNP\n",
       "2         rs3832852          2              A2M    DBSNP\n",
       "3             rs669          2              A2M      VEP\n",
       "4             rs669          2              A2M    DBSNP\n",
       "...             ...        ...              ...      ...\n",
       "361962  rs556554798  115072896  SLCO1B3-SLCO1B7      VEP\n",
       "361963   rs11045626  115072896  SLCO1B3-SLCO1B7      VEP\n",
       "361964    rs4149117  115072896  SLCO1B3-SLCO1B7      VEP\n",
       "361965     rs629444  115482690            H2BP5      VEP\n",
       "361966    rs3803700  115804232           CEROX1      VEP\n",
       "\n",
       "[361967 rows x 4 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Loading the CSV file into a DataFrame\n",
    "variant_to_gene_mappings = pd.read_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Data\\Protein\\Disgenet\\variant_to_gene_mappings.tsv', delimiter='\\t')\n",
    "variant_to_gene_mappings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8a77e51d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snpId</th>\n",
       "      <th>class</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>DSI</th>\n",
       "      <th>DPI</th>\n",
       "      <th>NofDiseases</th>\n",
       "      <th>NofPmids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000005</td>\n",
       "      <td>snv</td>\n",
       "      <td>21</td>\n",
       "      <td>33060745</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs10000770</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142693109</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1000096</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>38691214</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs10001106</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>10125817</td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>0.925</td>\n",
       "      <td>0.12</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194510</th>\n",
       "      <td>rs999943</td>\n",
       "      <td>snv</td>\n",
       "      <td>6</td>\n",
       "      <td>33656956</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194511</th>\n",
       "      <td>rs999944</td>\n",
       "      <td>snv</td>\n",
       "      <td>2</td>\n",
       "      <td>64822719</td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194512</th>\n",
       "      <td>rs999947969</td>\n",
       "      <td>snv</td>\n",
       "      <td>7</td>\n",
       "      <td>80671145</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>0.827</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194513</th>\n",
       "      <td>rs9999653</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>55632082</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194514</th>\n",
       "      <td>rs9999820</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>155197173</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>0.925</td>\n",
       "      <td>0.08</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>194515 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               snpId class chromosome   position most_severe_consequence  \\\n",
       "0          rs1000005   snv         21   33060745          intron variant   \n",
       "1         rs10000770   snv          4  142693109          intron variant   \n",
       "2       rs1000091588   snv         11   68935374             stop gained   \n",
       "3          rs1000096   snv          4   38691214          intron variant   \n",
       "4         rs10001106   snv          4   10125817      intergenic variant   \n",
       "...              ...   ...        ...        ...                     ...   \n",
       "194510      rs999943   snv          6   33656956          intron variant   \n",
       "194511      rs999944   snv          2   64822719      intergenic variant   \n",
       "194512   rs999947969   snv          7   80671145      synonymous variant   \n",
       "194513     rs9999653   snv          4   55632082          intron variant   \n",
       "194514     rs9999820   snv          4  155197173          intron variant   \n",
       "\n",
       "          DSI   DPI  NofDiseases  NofPmids  \n",
       "0         NaN   NaN            1         1  \n",
       "1       1.000  0.04            1         1  \n",
       "2       1.000  0.12            1         1  \n",
       "3         NaN   NaN            1         1  \n",
       "4       0.925  0.12            3         4  \n",
       "...       ...   ...          ...       ...  \n",
       "194510  1.000  0.08            1         1  \n",
       "194511  1.000  0.08            1         1  \n",
       "194512  0.827  0.08            7         7  \n",
       "194513  1.000  0.08            1         1  \n",
       "194514  0.925  0.08            2         1  \n",
       "\n",
       "[194515 rows x 9 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Loading the CSV file into a DataFrame\n",
    "variant_associations = pd.read_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Data\\Protein\\Disgenet\\variant_associations.tsv', delimiter='\\t')\n",
    "variant_associations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "db403292",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snpId</th>\n",
       "      <th>class</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>DSI</th>\n",
       "      <th>DPI</th>\n",
       "      <th>NofDiseases</th>\n",
       "      <th>NofPmids</th>\n",
       "      <th>geneId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>sourceId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000005</td>\n",
       "      <td>snv</td>\n",
       "      <td>21</td>\n",
       "      <td>33060745</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>101930746</td>\n",
       "      <td>LINC00945</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs10000770</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142693109</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8821</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs10000770</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142693109</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8821</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3508</td>\n",
       "      <td>IGHMBP2</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3508</td>\n",
       "      <td>IGHMBP2</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361962</th>\n",
       "      <td>rs999947969</td>\n",
       "      <td>snv</td>\n",
       "      <td>7</td>\n",
       "      <td>80671145</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>0.827</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>948</td>\n",
       "      <td>CD36</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361963</th>\n",
       "      <td>rs999947969</td>\n",
       "      <td>snv</td>\n",
       "      <td>7</td>\n",
       "      <td>80671145</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>0.827</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>948</td>\n",
       "      <td>CD36</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361964</th>\n",
       "      <td>rs9999653</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>55632082</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>10874</td>\n",
       "      <td>NMU</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361965</th>\n",
       "      <td>rs9999653</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>55632082</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>10874</td>\n",
       "      <td>NMU</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361966</th>\n",
       "      <td>rs9999820</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>155197173</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>0.925</td>\n",
       "      <td>0.08</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>4887</td>\n",
       "      <td>NPY2R</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>361967 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               snpId class chromosome   position most_severe_consequence  \\\n",
       "0          rs1000005   snv         21   33060745          intron variant   \n",
       "1         rs10000770   snv          4  142693109          intron variant   \n",
       "2         rs10000770   snv          4  142693109          intron variant   \n",
       "3       rs1000091588   snv         11   68935374             stop gained   \n",
       "4       rs1000091588   snv         11   68935374             stop gained   \n",
       "...              ...   ...        ...        ...                     ...   \n",
       "361962   rs999947969   snv          7   80671145      synonymous variant   \n",
       "361963   rs999947969   snv          7   80671145      synonymous variant   \n",
       "361964     rs9999653   snv          4   55632082          intron variant   \n",
       "361965     rs9999653   snv          4   55632082          intron variant   \n",
       "361966     rs9999820   snv          4  155197173          intron variant   \n",
       "\n",
       "          DSI   DPI  NofDiseases  NofPmids     geneId geneSymbol sourceId  \n",
       "0         NaN   NaN            1         1  101930746  LINC00945    DBSNP  \n",
       "1       1.000  0.04            1         1       8821     INPP4B    DBSNP  \n",
       "2       1.000  0.04            1         1       8821     INPP4B      VEP  \n",
       "3       1.000  0.12            1         1       3508    IGHMBP2      VEP  \n",
       "4       1.000  0.12            1         1       3508    IGHMBP2    DBSNP  \n",
       "...       ...   ...          ...       ...        ...        ...      ...  \n",
       "361962  0.827  0.08            7         7        948       CD36    DBSNP  \n",
       "361963  0.827  0.08            7         7        948       CD36      VEP  \n",
       "361964  1.000  0.08            1         1      10874        NMU      VEP  \n",
       "361965  1.000  0.08            1         1      10874        NMU    DBSNP  \n",
       "361966  0.925  0.08            2         1       4887      NPY2R    DBSNP  \n",
       "\n",
       "[361967 rows x 12 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merging 'variant_associations' and 'variant_to_gene_mappings'\n",
    "Disgenet_merged = pd.merge(variant_associations, variant_to_gene_mappings, how='inner', on='snpId')\n",
    "Disgenet_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "65f4720c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snpId</th>\n",
       "      <th>class</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>DSI_x</th>\n",
       "      <th>DPI_x</th>\n",
       "      <th>NofDiseases_x</th>\n",
       "      <th>NofPmids_x</th>\n",
       "      <th>geneId</th>\n",
       "      <th>geneSymbol_x</th>\n",
       "      <th>sourceId</th>\n",
       "      <th>geneSymbol_y</th>\n",
       "      <th>DSI_y</th>\n",
       "      <th>DPI_y</th>\n",
       "      <th>PLI</th>\n",
       "      <th>protein_class_name</th>\n",
       "      <th>protein_class</th>\n",
       "      <th>NofDiseases_y</th>\n",
       "      <th>NofPmids_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs10000770</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142693109</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8821</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.573</td>\n",
       "      <td>0.577</td>\n",
       "      <td>0.552180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs10000770</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142693109</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8821</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>VEP</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.573</td>\n",
       "      <td>0.577</td>\n",
       "      <td>0.552180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs10006418</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142691588</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8821</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>VEP</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.573</td>\n",
       "      <td>0.577</td>\n",
       "      <td>0.552180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs10006418</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142691588</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8821</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.573</td>\n",
       "      <td>0.577</td>\n",
       "      <td>0.552180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs10012488</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142671933</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8821</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>VEP</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.573</td>\n",
       "      <td>0.577</td>\n",
       "      <td>0.552180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346532</th>\n",
       "      <td>rs998297769</td>\n",
       "      <td>snv</td>\n",
       "      <td>X</td>\n",
       "      <td>151962587</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2564</td>\n",
       "      <td>GABRE</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>GABRE</td>\n",
       "      <td>0.839</td>\n",
       "      <td>0.231</td>\n",
       "      <td>0.000184</td>\n",
       "      <td>Ion channel</td>\n",
       "      <td>DTO_01300327</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346533</th>\n",
       "      <td>rs999036825</td>\n",
       "      <td>snv</td>\n",
       "      <td>19</td>\n",
       "      <td>16228789</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>8907</td>\n",
       "      <td>AP1M1</td>\n",
       "      <td>VEP</td>\n",
       "      <td>AP1M1</td>\n",
       "      <td>0.839</td>\n",
       "      <td>0.154</td>\n",
       "      <td>0.712510</td>\n",
       "      <td>Extracellular structure</td>\n",
       "      <td>DTO_05007621</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346534</th>\n",
       "      <td>rs999036825</td>\n",
       "      <td>snv</td>\n",
       "      <td>19</td>\n",
       "      <td>16228789</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>8907</td>\n",
       "      <td>AP1M1</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>AP1M1</td>\n",
       "      <td>0.839</td>\n",
       "      <td>0.154</td>\n",
       "      <td>0.712510</td>\n",
       "      <td>Extracellular structure</td>\n",
       "      <td>DTO_05007621</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346535</th>\n",
       "      <td>rs9992013</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>2257736</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>10608</td>\n",
       "      <td>MXD4</td>\n",
       "      <td>VEP</td>\n",
       "      <td>MXD4</td>\n",
       "      <td>0.722</td>\n",
       "      <td>0.115</td>\n",
       "      <td>0.776350</td>\n",
       "      <td>Transcription factor</td>\n",
       "      <td>DTO_05007542</td>\n",
       "      <td>19</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346536</th>\n",
       "      <td>rs9992013</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>2257736</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>10608</td>\n",
       "      <td>MXD4</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>MXD4</td>\n",
       "      <td>0.722</td>\n",
       "      <td>0.115</td>\n",
       "      <td>0.776350</td>\n",
       "      <td>Transcription factor</td>\n",
       "      <td>DTO_05007542</td>\n",
       "      <td>19</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>346537 rows Ã— 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              snpId class chromosome   position most_severe_consequence  \\\n",
       "0        rs10000770   snv          4  142693109          intron variant   \n",
       "1        rs10000770   snv          4  142693109          intron variant   \n",
       "2        rs10006418   snv          4  142691588          intron variant   \n",
       "3        rs10006418   snv          4  142691588          intron variant   \n",
       "4        rs10012488   snv          4  142671933          intron variant   \n",
       "...             ...   ...        ...        ...                     ...   \n",
       "346532  rs998297769   snv          X  151962587             stop gained   \n",
       "346533  rs999036825   snv         19   16228789        missense variant   \n",
       "346534  rs999036825   snv         19   16228789        missense variant   \n",
       "346535    rs9992013   snv          4    2257736          intron variant   \n",
       "346536    rs9992013   snv          4    2257736          intron variant   \n",
       "\n",
       "        DSI_x  DPI_x  NofDiseases_x  NofPmids_x  geneId geneSymbol_x sourceId  \\\n",
       "0         1.0   0.04              1           1    8821       INPP4B    DBSNP   \n",
       "1         1.0   0.04              1           1    8821       INPP4B      VEP   \n",
       "2         1.0   0.04              1           1    8821       INPP4B      VEP   \n",
       "3         1.0   0.04              1           1    8821       INPP4B    DBSNP   \n",
       "4         1.0   0.04              1           1    8821       INPP4B      VEP   \n",
       "...       ...    ...            ...         ...     ...          ...      ...   \n",
       "346532    NaN    NaN              1           0    2564        GABRE    DBSNP   \n",
       "346533    1.0   0.08              1           0    8907        AP1M1      VEP   \n",
       "346534    1.0   0.08              1           0    8907        AP1M1    DBSNP   \n",
       "346535    NaN    NaN              1           1   10608         MXD4      VEP   \n",
       "346536    NaN    NaN              1           1   10608         MXD4    DBSNP   \n",
       "\n",
       "       geneSymbol_y  DSI_y  DPI_y       PLI       protein_class_name  \\\n",
       "0            INPP4B  0.573  0.577  0.552180                      NaN   \n",
       "1            INPP4B  0.573  0.577  0.552180                      NaN   \n",
       "2            INPP4B  0.573  0.577  0.552180                      NaN   \n",
       "3            INPP4B  0.573  0.577  0.552180                      NaN   \n",
       "4            INPP4B  0.573  0.577  0.552180                      NaN   \n",
       "...             ...    ...    ...       ...                      ...   \n",
       "346532        GABRE  0.839  0.231  0.000184              Ion channel   \n",
       "346533        AP1M1  0.839  0.154  0.712510  Extracellular structure   \n",
       "346534        AP1M1  0.839  0.154  0.712510  Extracellular structure   \n",
       "346535         MXD4  0.722  0.115  0.776350     Transcription factor   \n",
       "346536         MXD4  0.722  0.115  0.776350     Transcription factor   \n",
       "\n",
       "       protein_class  NofDiseases_y  NofPmids_y  \n",
       "0                NaN             90          72  \n",
       "1                NaN             90          72  \n",
       "2                NaN             90          72  \n",
       "3                NaN             90          72  \n",
       "4                NaN             90          72  \n",
       "...              ...            ...         ...  \n",
       "346532  DTO_01300327              8           4  \n",
       "346533  DTO_05007621              5           1  \n",
       "346534  DTO_05007621              5           1  \n",
       "346535  DTO_05007542             19           9  \n",
       "346536  DTO_05007542             19           9  \n",
       "\n",
       "[346537 rows x 20 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merging 'variant_associations+variant_to_gene_mappings' and 'gene_associations'\n",
    "Disgenet_all_merged = pd.merge(Disgenet_merged, gene_associations, how='inner', on='geneId')\n",
    "Disgenet_all_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f7a60b4",
   "metadata": {},
   "source": [
    "### Data Harmonizome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1c7e9a7e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSym</th>\n",
       "      <th>NA</th>\n",
       "      <th>GeneID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PITPNM2</td>\n",
       "      <td>na</td>\n",
       "      <td>57605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VRK1</td>\n",
       "      <td>na</td>\n",
       "      <td>7443</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OGFOD2</td>\n",
       "      <td>na</td>\n",
       "      <td>79676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KIDINS220</td>\n",
       "      <td>na</td>\n",
       "      <td>57498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IGSF10</td>\n",
       "      <td>na</td>\n",
       "      <td>285313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10700</th>\n",
       "      <td>EBAG9</td>\n",
       "      <td>na</td>\n",
       "      <td>9166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10701</th>\n",
       "      <td>IGHV1-69</td>\n",
       "      <td>na</td>\n",
       "      <td>28461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10702</th>\n",
       "      <td>ERAL1</td>\n",
       "      <td>na</td>\n",
       "      <td>26284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10703</th>\n",
       "      <td>SLC27A5</td>\n",
       "      <td>na</td>\n",
       "      <td>10998</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10704</th>\n",
       "      <td>CD177</td>\n",
       "      <td>na</td>\n",
       "      <td>57126</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10705 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         GeneSym  NA  GeneID\n",
       "0        PITPNM2  na   57605\n",
       "1           VRK1  na    7443\n",
       "2         OGFOD2  na   79676\n",
       "3      KIDINS220  na   57498\n",
       "4         IGSF10  na  285313\n",
       "...          ...  ..     ...\n",
       "10700      EBAG9  na    9166\n",
       "10701   IGHV1-69  na   28461\n",
       "10702      ERAL1  na   26284\n",
       "10703    SLC27A5  na   10998\n",
       "10704      CD177  na   57126\n",
       "\n",
       "[10705 rows x 3 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Loading the CSV file into a DataFrame\n",
    "gene_list_terms = pd.read_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Data\\Protein\\Harmonizome 3.0\\gene_list_terms.txt', delimiter='\\t')\n",
    "gene_list_terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "16862f1e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Loading the CSV file into a DataFrame\n",
    "# gene_set_library_crisp = pd.read_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Data\\Protein\\Harmonizome 3.0\\gene_set_library_crisp.txt', delimiter='\\t')\n",
    "# gene_set_library_crisp\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "094e96df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                 Column1          Column2  \\\n",
      "0                          eating behavior disinhibition        metabolic   \n",
      "1                        phenylthiocarbamide sensitivity  normalvariation   \n",
      "2                    bitter perception; sweet preference  normalvariation   \n",
      "3                   ability to taste phenylthiocarbamide            other   \n",
      "4                                          dental caries        infection   \n",
      "...                                                  ...              ...   \n",
      "12770                            mucopolysaccharidosis i            other   \n",
      "12771                          hypertension;  benazepril          unknown   \n",
      "12772     congenital fibrosis of the extraocular muscles    developmental   \n",
      "12773  meckel-gruber syndrome; abnormalities, multipl...          unknown   \n",
      "12774  lung diseases, interstitial; scleroderma, diff...          unknown   \n",
      "\n",
      "        Column3  \n",
      "0       TAS2R38  \n",
      "1       TAS2R38  \n",
      "2       TAS2R38  \n",
      "3       TAS2R38  \n",
      "4       TAS2R38  \n",
      "...         ...  \n",
      "12770      IDUA  \n",
      "12771      PRCP  \n",
      "12772    KIF21A  \n",
      "12773      MKS1  \n",
      "12774  TRBV20-1  \n",
      "\n",
      "[12775 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "# Empty list to store the data\n",
    "data = []\n",
    "\n",
    "# Open the file and read all the lines\n",
    "with open(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Data\\Protein\\Harmonizome 3.0\\gene_set_library_crisp.txt') as file:\n",
    "    for line in file:\n",
    "        # Split the line values and add to the data list\n",
    "        split_line = line.strip().split('\\t')\n",
    "        data.append([split_line[0], split_line[1], split_line[2]])\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(data, columns=['Column1', 'Column2', 'Column3'])\n",
    "\n",
    "# Display DataFrame\n",
    "print(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "798d9ec5",
   "metadata": {},
   "outputs": [],
   "source": [
    "#saving the DataFrame to a CSV file \n",
    "df.to_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Data\\Protein\\Harmonizome 3.0\\df.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "99d01dd5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Column1</th>\n",
       "      <th>Column2</th>\n",
       "      <th>Column3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>eating behavior disinhibition</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>TAS2R38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>phenylthiocarbamide sensitivity</td>\n",
       "      <td>normalvariation</td>\n",
       "      <td>TAS2R38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>bitter perception; sweet preference</td>\n",
       "      <td>normalvariation</td>\n",
       "      <td>TAS2R38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ability to taste phenylthiocarbamide</td>\n",
       "      <td>other</td>\n",
       "      <td>TAS2R38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>dental caries</td>\n",
       "      <td>infection</td>\n",
       "      <td>TAS2R38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12770</th>\n",
       "      <td>mucopolysaccharidosis i</td>\n",
       "      <td>other</td>\n",
       "      <td>IDUA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12771</th>\n",
       "      <td>hypertension;  benazepril</td>\n",
       "      <td>unknown</td>\n",
       "      <td>PRCP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12772</th>\n",
       "      <td>congenital fibrosis of the extraocular muscles</td>\n",
       "      <td>developmental</td>\n",
       "      <td>KIF21A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12773</th>\n",
       "      <td>meckel-gruber syndrome; abnormalities, multipl...</td>\n",
       "      <td>unknown</td>\n",
       "      <td>MKS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12774</th>\n",
       "      <td>lung diseases, interstitial; scleroderma, diff...</td>\n",
       "      <td>unknown</td>\n",
       "      <td>TRBV20-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12775 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 Column1          Column2  \\\n",
       "0                          eating behavior disinhibition        metabolic   \n",
       "1                        phenylthiocarbamide sensitivity  normalvariation   \n",
       "2                    bitter perception; sweet preference  normalvariation   \n",
       "3                   ability to taste phenylthiocarbamide            other   \n",
       "4                                          dental caries        infection   \n",
       "...                                                  ...              ...   \n",
       "12770                            mucopolysaccharidosis i            other   \n",
       "12771                          hypertension;  benazepril          unknown   \n",
       "12772     congenital fibrosis of the extraocular muscles    developmental   \n",
       "12773  meckel-gruber syndrome; abnormalities, multipl...          unknown   \n",
       "12774  lung diseases, interstitial; scleroderma, diff...          unknown   \n",
       "\n",
       "        Column3  \n",
       "0       TAS2R38  \n",
       "1       TAS2R38  \n",
       "2       TAS2R38  \n",
       "3       TAS2R38  \n",
       "4       TAS2R38  \n",
       "...         ...  \n",
       "12770      IDUA  \n",
       "12771      PRCP  \n",
       "12772    KIF21A  \n",
       "12773      MKS1  \n",
       "12774  TRBV20-1  \n",
       "\n",
       "[12775 rows x 3 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Loading the CSV file into a DataFrame\n",
    "gene_set_library_crisp = pd.read_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Data\\Protein\\Harmonizome 3.0\\df.csv')\n",
    "gene_set_library_crisp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "73d3e41c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        GeneSym  NA  GeneID  \\\n",
      "0       PITPNM2  na   57605   \n",
      "1         ABCB9  na   23457   \n",
      "2          CROT  na   54677   \n",
      "3        FBXO11  na   80204   \n",
      "4        FBXO11  na   80204   \n",
      "...         ...  ..     ...   \n",
      "12770     EBAG9  na    9166   \n",
      "12771  IGHV1-69  na   28461   \n",
      "12772     ERAL1  na   26284   \n",
      "12773   SLC27A5  na   10998   \n",
      "12774     CD177  na   57126   \n",
      "\n",
      "                                                 Column1          Column2  \\\n",
      "0                                            neutrophils           immune   \n",
      "1                                     drug-related genes  pharmacogenomic   \n",
      "2                                   tobacco use disorder          unknown   \n",
      "3                                               vitiligo            other   \n",
      "4                                            lymphocytes           immune   \n",
      "...                                                  ...              ...   \n",
      "12770                         early-stage breast cancers           cancer   \n",
      "12771  leukemia, lymphocytic, chronic, b-cell; lympho...           cancer   \n",
      "12772                skeletal responsiveness to estrogen        metabolic   \n",
      "12773  fatty liver; insulin resistance; metabolic syn...        metabolic   \n",
      "12774                              low expression of nb1            other   \n",
      "\n",
      "        Column3  \n",
      "0       PITPNM2  \n",
      "1         ABCB9  \n",
      "2          CROT  \n",
      "3        FBXO11  \n",
      "4        FBXO11  \n",
      "...         ...  \n",
      "12770     EBAG9  \n",
      "12771  IGHV1-69  \n",
      "12772     ERAL1  \n",
      "12773   SLC27A5  \n",
      "12774     CD177  \n",
      "\n",
      "[12775 rows x 6 columns]\n"
     ]
    }
   ],
   "source": [
    "#Merging 'gene_list_terms' and 'gene_set_library_crisp'\n",
    "merged_df = pd.merge(gene_list_terms, gene_set_library_crisp, how='inner', left_on='GeneSym', right_on='Column3')\n",
    "print(merged_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "4c11c381",
   "metadata": {},
   "outputs": [],
   "source": [
    "#saving the DataFrame to a CSV file \n",
    "merged_df.to_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Data\\Protein\\Harmonizome 3.0\\merged.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "71adc03a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSym</th>\n",
       "      <th>NA</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Column1</th>\n",
       "      <th>Column2</th>\n",
       "      <th>Column3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PITPNM2</td>\n",
       "      <td>na</td>\n",
       "      <td>57605</td>\n",
       "      <td>neutrophils</td>\n",
       "      <td>immune</td>\n",
       "      <td>PITPNM2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCB9</td>\n",
       "      <td>na</td>\n",
       "      <td>23457</td>\n",
       "      <td>drug-related genes</td>\n",
       "      <td>pharmacogenomic</td>\n",
       "      <td>ABCB9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CROT</td>\n",
       "      <td>na</td>\n",
       "      <td>54677</td>\n",
       "      <td>tobacco use disorder</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FBXO11</td>\n",
       "      <td>na</td>\n",
       "      <td>80204</td>\n",
       "      <td>vitiligo</td>\n",
       "      <td>other</td>\n",
       "      <td>FBXO11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FBXO11</td>\n",
       "      <td>na</td>\n",
       "      <td>80204</td>\n",
       "      <td>lymphocytes</td>\n",
       "      <td>immune</td>\n",
       "      <td>FBXO11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12770</th>\n",
       "      <td>EBAG9</td>\n",
       "      <td>na</td>\n",
       "      <td>9166</td>\n",
       "      <td>early-stage breast cancers</td>\n",
       "      <td>cancer</td>\n",
       "      <td>EBAG9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12771</th>\n",
       "      <td>IGHV1-69</td>\n",
       "      <td>na</td>\n",
       "      <td>28461</td>\n",
       "      <td>leukemia, lymphocytic, chronic, b-cell; lympho...</td>\n",
       "      <td>cancer</td>\n",
       "      <td>IGHV1-69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12772</th>\n",
       "      <td>ERAL1</td>\n",
       "      <td>na</td>\n",
       "      <td>26284</td>\n",
       "      <td>skeletal responsiveness to estrogen</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>ERAL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12773</th>\n",
       "      <td>SLC27A5</td>\n",
       "      <td>na</td>\n",
       "      <td>10998</td>\n",
       "      <td>fatty liver; insulin resistance; metabolic syn...</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>SLC27A5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12774</th>\n",
       "      <td>CD177</td>\n",
       "      <td>na</td>\n",
       "      <td>57126</td>\n",
       "      <td>low expression of nb1</td>\n",
       "      <td>other</td>\n",
       "      <td>CD177</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12775 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GeneSym  NA  GeneID  \\\n",
       "0       PITPNM2  na   57605   \n",
       "1         ABCB9  na   23457   \n",
       "2          CROT  na   54677   \n",
       "3        FBXO11  na   80204   \n",
       "4        FBXO11  na   80204   \n",
       "...         ...  ..     ...   \n",
       "12770     EBAG9  na    9166   \n",
       "12771  IGHV1-69  na   28461   \n",
       "12772     ERAL1  na   26284   \n",
       "12773   SLC27A5  na   10998   \n",
       "12774     CD177  na   57126   \n",
       "\n",
       "                                                 Column1          Column2  \\\n",
       "0                                            neutrophils           immune   \n",
       "1                                     drug-related genes  pharmacogenomic   \n",
       "2                                   tobacco use disorder          unknown   \n",
       "3                                               vitiligo            other   \n",
       "4                                            lymphocytes           immune   \n",
       "...                                                  ...              ...   \n",
       "12770                         early-stage breast cancers           cancer   \n",
       "12771  leukemia, lymphocytic, chronic, b-cell; lympho...           cancer   \n",
       "12772                skeletal responsiveness to estrogen        metabolic   \n",
       "12773  fatty liver; insulin resistance; metabolic syn...        metabolic   \n",
       "12774                              low expression of nb1            other   \n",
       "\n",
       "        Column3  \n",
       "0       PITPNM2  \n",
       "1         ABCB9  \n",
       "2          CROT  \n",
       "3        FBXO11  \n",
       "4        FBXO11  \n",
       "...         ...  \n",
       "12770     EBAG9  \n",
       "12771  IGHV1-69  \n",
       "12772     ERAL1  \n",
       "12773   SLC27A5  \n",
       "12774     CD177  \n",
       "\n",
       "[12775 rows x 6 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Harmonizome = pd.read_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Data\\Protein\\Harmonizome 3.0\\merged.csv')\n",
    "Harmonizome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "15f4192d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Column1</th>\n",
       "      <th>Column2</th>\n",
       "      <th>Column3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>57605</td>\n",
       "      <td>neutrophils</td>\n",
       "      <td>immune</td>\n",
       "      <td>PITPNM2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23457</td>\n",
       "      <td>drug-related genes</td>\n",
       "      <td>pharmacogenomic</td>\n",
       "      <td>ABCB9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>54677</td>\n",
       "      <td>tobacco use disorder</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>80204</td>\n",
       "      <td>vitiligo</td>\n",
       "      <td>other</td>\n",
       "      <td>FBXO11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>80204</td>\n",
       "      <td>lymphocytes</td>\n",
       "      <td>immune</td>\n",
       "      <td>FBXO11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12770</th>\n",
       "      <td>9166</td>\n",
       "      <td>early-stage breast cancers</td>\n",
       "      <td>cancer</td>\n",
       "      <td>EBAG9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12771</th>\n",
       "      <td>28461</td>\n",
       "      <td>leukemia, lymphocytic, chronic, b-cell; lympho...</td>\n",
       "      <td>cancer</td>\n",
       "      <td>IGHV1-69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12772</th>\n",
       "      <td>26284</td>\n",
       "      <td>skeletal responsiveness to estrogen</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>ERAL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12773</th>\n",
       "      <td>10998</td>\n",
       "      <td>fatty liver; insulin resistance; metabolic syn...</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>SLC27A5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12774</th>\n",
       "      <td>57126</td>\n",
       "      <td>low expression of nb1</td>\n",
       "      <td>other</td>\n",
       "      <td>CD177</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12775 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneID                                            Column1  \\\n",
       "0       57605                                        neutrophils   \n",
       "1       23457                                 drug-related genes   \n",
       "2       54677                               tobacco use disorder   \n",
       "3       80204                                           vitiligo   \n",
       "4       80204                                        lymphocytes   \n",
       "...       ...                                                ...   \n",
       "12770    9166                         early-stage breast cancers   \n",
       "12771   28461  leukemia, lymphocytic, chronic, b-cell; lympho...   \n",
       "12772   26284                skeletal responsiveness to estrogen   \n",
       "12773   10998  fatty liver; insulin resistance; metabolic syn...   \n",
       "12774   57126                              low expression of nb1   \n",
       "\n",
       "               Column2   Column3  \n",
       "0               immune   PITPNM2  \n",
       "1      pharmacogenomic     ABCB9  \n",
       "2              unknown      CROT  \n",
       "3                other    FBXO11  \n",
       "4               immune    FBXO11  \n",
       "...                ...       ...  \n",
       "12770           cancer     EBAG9  \n",
       "12771           cancer  IGHV1-69  \n",
       "12772        metabolic     ERAL1  \n",
       "12773        metabolic   SLC27A5  \n",
       "12774            other     CD177  \n",
       "\n",
       "[12775 rows x 4 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Dropping columns  \n",
    "Harmonizome = Harmonizome.drop(columns=['NA','GeneSym'])\n",
    "Harmonizome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "20bec9b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Disease Class</th>\n",
       "      <th>Category</th>\n",
       "      <th>Gene/Protein Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>57605</td>\n",
       "      <td>neutrophils</td>\n",
       "      <td>immune</td>\n",
       "      <td>PITPNM2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23457</td>\n",
       "      <td>drug-related genes</td>\n",
       "      <td>pharmacogenomic</td>\n",
       "      <td>ABCB9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>54677</td>\n",
       "      <td>tobacco use disorder</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>80204</td>\n",
       "      <td>vitiligo</td>\n",
       "      <td>other</td>\n",
       "      <td>FBXO11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>80204</td>\n",
       "      <td>lymphocytes</td>\n",
       "      <td>immune</td>\n",
       "      <td>FBXO11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12770</th>\n",
       "      <td>9166</td>\n",
       "      <td>early-stage breast cancers</td>\n",
       "      <td>cancer</td>\n",
       "      <td>EBAG9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12771</th>\n",
       "      <td>28461</td>\n",
       "      <td>leukemia, lymphocytic, chronic, b-cell; lympho...</td>\n",
       "      <td>cancer</td>\n",
       "      <td>IGHV1-69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12772</th>\n",
       "      <td>26284</td>\n",
       "      <td>skeletal responsiveness to estrogen</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>ERAL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12773</th>\n",
       "      <td>10998</td>\n",
       "      <td>fatty liver; insulin resistance; metabolic syn...</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>SLC27A5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12774</th>\n",
       "      <td>57126</td>\n",
       "      <td>low expression of nb1</td>\n",
       "      <td>other</td>\n",
       "      <td>CD177</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12775 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneID                                      Disease Class  \\\n",
       "0       57605                                        neutrophils   \n",
       "1       23457                                 drug-related genes   \n",
       "2       54677                               tobacco use disorder   \n",
       "3       80204                                           vitiligo   \n",
       "4       80204                                        lymphocytes   \n",
       "...       ...                                                ...   \n",
       "12770    9166                         early-stage breast cancers   \n",
       "12771   28461  leukemia, lymphocytic, chronic, b-cell; lympho...   \n",
       "12772   26284                skeletal responsiveness to estrogen   \n",
       "12773   10998  fatty liver; insulin resistance; metabolic syn...   \n",
       "12774   57126                              low expression of nb1   \n",
       "\n",
       "              Category Gene/Protein Symbol  \n",
       "0               immune             PITPNM2  \n",
       "1      pharmacogenomic               ABCB9  \n",
       "2              unknown                CROT  \n",
       "3                other              FBXO11  \n",
       "4               immune              FBXO11  \n",
       "...                ...                 ...  \n",
       "12770           cancer               EBAG9  \n",
       "12771           cancer            IGHV1-69  \n",
       "12772        metabolic               ERAL1  \n",
       "12773        metabolic             SLC27A5  \n",
       "12774            other               CD177  \n",
       "\n",
       "[12775 rows x 4 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Renaming columns\n",
    "df = Harmonizome.rename(columns={'Column1': 'Disease Class', 'Column2': 'Category', 'Column3': 'Gene/Protein Symbol'})\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "e549125b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#creating a .csv file\n",
    "df.to_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Data\\Protein\\Harmonizome 3.0\\Harmonizome.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "26223bb6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Disease Class</th>\n",
       "      <th>Category</th>\n",
       "      <th>Gene/Protein Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>57605</td>\n",
       "      <td>neutrophils</td>\n",
       "      <td>immune</td>\n",
       "      <td>PITPNM2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23457</td>\n",
       "      <td>drug-related genes</td>\n",
       "      <td>pharmacogenomic</td>\n",
       "      <td>ABCB9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>54677</td>\n",
       "      <td>tobacco use disorder</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>80204</td>\n",
       "      <td>vitiligo</td>\n",
       "      <td>other</td>\n",
       "      <td>FBXO11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>80204</td>\n",
       "      <td>lymphocytes</td>\n",
       "      <td>immune</td>\n",
       "      <td>FBXO11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12770</th>\n",
       "      <td>9166</td>\n",
       "      <td>early-stage breast cancers</td>\n",
       "      <td>cancer</td>\n",
       "      <td>EBAG9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12771</th>\n",
       "      <td>28461</td>\n",
       "      <td>leukemia, lymphocytic, chronic, b-cell; lympho...</td>\n",
       "      <td>cancer</td>\n",
       "      <td>IGHV1-69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12772</th>\n",
       "      <td>26284</td>\n",
       "      <td>skeletal responsiveness to estrogen</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>ERAL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12773</th>\n",
       "      <td>10998</td>\n",
       "      <td>fatty liver; insulin resistance; metabolic syn...</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>SLC27A5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12774</th>\n",
       "      <td>57126</td>\n",
       "      <td>low expression of nb1</td>\n",
       "      <td>other</td>\n",
       "      <td>CD177</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12775 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneID                                      Disease Class  \\\n",
       "0       57605                                        neutrophils   \n",
       "1       23457                                 drug-related genes   \n",
       "2       54677                               tobacco use disorder   \n",
       "3       80204                                           vitiligo   \n",
       "4       80204                                        lymphocytes   \n",
       "...       ...                                                ...   \n",
       "12770    9166                         early-stage breast cancers   \n",
       "12771   28461  leukemia, lymphocytic, chronic, b-cell; lympho...   \n",
       "12772   26284                skeletal responsiveness to estrogen   \n",
       "12773   10998  fatty liver; insulin resistance; metabolic syn...   \n",
       "12774   57126                              low expression of nb1   \n",
       "\n",
       "              Category Gene/Protein Symbol  \n",
       "0               immune             PITPNM2  \n",
       "1      pharmacogenomic               ABCB9  \n",
       "2              unknown                CROT  \n",
       "3                other              FBXO11  \n",
       "4               immune              FBXO11  \n",
       "...                ...                 ...  \n",
       "12770           cancer               EBAG9  \n",
       "12771           cancer            IGHV1-69  \n",
       "12772        metabolic               ERAL1  \n",
       "12773        metabolic             SLC27A5  \n",
       "12774            other               CD177  \n",
       "\n",
       "[12775 rows x 4 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Loading the CSV file into a DataFrame\n",
    "df_Harmonizome = pd.read_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Data\\Protein\\Harmonizome 3.0\\Harmonizome.csv')\n",
    "df_Harmonizome"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2691af13",
   "metadata": {},
   "source": [
    "### Merge the df's - DisGeNET + Harmonizome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "4e0b02b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snpId</th>\n",
       "      <th>class</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>DSI_x</th>\n",
       "      <th>DPI_x</th>\n",
       "      <th>NofDiseases_x</th>\n",
       "      <th>NofPmids_x</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>geneSymbol_x</th>\n",
       "      <th>sourceId</th>\n",
       "      <th>geneSymbol_y</th>\n",
       "      <th>DSI_y</th>\n",
       "      <th>DPI_y</th>\n",
       "      <th>PLI</th>\n",
       "      <th>protein_class_name</th>\n",
       "      <th>protein_class</th>\n",
       "      <th>NofDiseases_y</th>\n",
       "      <th>NofPmids_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs10000770</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142693109</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8821</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.573</td>\n",
       "      <td>0.577</td>\n",
       "      <td>0.552180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs10000770</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142693109</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8821</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>VEP</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.573</td>\n",
       "      <td>0.577</td>\n",
       "      <td>0.552180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs10006418</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142691588</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8821</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>VEP</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.573</td>\n",
       "      <td>0.577</td>\n",
       "      <td>0.552180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs10006418</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142691588</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8821</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.573</td>\n",
       "      <td>0.577</td>\n",
       "      <td>0.552180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs10012488</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>142671933</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8821</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>VEP</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.573</td>\n",
       "      <td>0.577</td>\n",
       "      <td>0.552180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346532</th>\n",
       "      <td>rs998297769</td>\n",
       "      <td>snv</td>\n",
       "      <td>X</td>\n",
       "      <td>151962587</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2564</td>\n",
       "      <td>GABRE</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>GABRE</td>\n",
       "      <td>0.839</td>\n",
       "      <td>0.231</td>\n",
       "      <td>0.000184</td>\n",
       "      <td>Ion channel</td>\n",
       "      <td>DTO_01300327</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346533</th>\n",
       "      <td>rs999036825</td>\n",
       "      <td>snv</td>\n",
       "      <td>19</td>\n",
       "      <td>16228789</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>8907</td>\n",
       "      <td>AP1M1</td>\n",
       "      <td>VEP</td>\n",
       "      <td>AP1M1</td>\n",
       "      <td>0.839</td>\n",
       "      <td>0.154</td>\n",
       "      <td>0.712510</td>\n",
       "      <td>Extracellular structure</td>\n",
       "      <td>DTO_05007621</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346534</th>\n",
       "      <td>rs999036825</td>\n",
       "      <td>snv</td>\n",
       "      <td>19</td>\n",
       "      <td>16228789</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>8907</td>\n",
       "      <td>AP1M1</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>AP1M1</td>\n",
       "      <td>0.839</td>\n",
       "      <td>0.154</td>\n",
       "      <td>0.712510</td>\n",
       "      <td>Extracellular structure</td>\n",
       "      <td>DTO_05007621</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346535</th>\n",
       "      <td>rs9992013</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>2257736</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>10608</td>\n",
       "      <td>MXD4</td>\n",
       "      <td>VEP</td>\n",
       "      <td>MXD4</td>\n",
       "      <td>0.722</td>\n",
       "      <td>0.115</td>\n",
       "      <td>0.776350</td>\n",
       "      <td>Transcription factor</td>\n",
       "      <td>DTO_05007542</td>\n",
       "      <td>19</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346536</th>\n",
       "      <td>rs9992013</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>2257736</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>10608</td>\n",
       "      <td>MXD4</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>MXD4</td>\n",
       "      <td>0.722</td>\n",
       "      <td>0.115</td>\n",
       "      <td>0.776350</td>\n",
       "      <td>Transcription factor</td>\n",
       "      <td>DTO_05007542</td>\n",
       "      <td>19</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>346537 rows Ã— 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              snpId class chromosome   position most_severe_consequence  \\\n",
       "0        rs10000770   snv          4  142693109          intron variant   \n",
       "1        rs10000770   snv          4  142693109          intron variant   \n",
       "2        rs10006418   snv          4  142691588          intron variant   \n",
       "3        rs10006418   snv          4  142691588          intron variant   \n",
       "4        rs10012488   snv          4  142671933          intron variant   \n",
       "...             ...   ...        ...        ...                     ...   \n",
       "346532  rs998297769   snv          X  151962587             stop gained   \n",
       "346533  rs999036825   snv         19   16228789        missense variant   \n",
       "346534  rs999036825   snv         19   16228789        missense variant   \n",
       "346535    rs9992013   snv          4    2257736          intron variant   \n",
       "346536    rs9992013   snv          4    2257736          intron variant   \n",
       "\n",
       "        DSI_x  DPI_x  NofDiseases_x  NofPmids_x  GeneID geneSymbol_x sourceId  \\\n",
       "0         1.0   0.04              1           1    8821       INPP4B    DBSNP   \n",
       "1         1.0   0.04              1           1    8821       INPP4B      VEP   \n",
       "2         1.0   0.04              1           1    8821       INPP4B      VEP   \n",
       "3         1.0   0.04              1           1    8821       INPP4B    DBSNP   \n",
       "4         1.0   0.04              1           1    8821       INPP4B      VEP   \n",
       "...       ...    ...            ...         ...     ...          ...      ...   \n",
       "346532    NaN    NaN              1           0    2564        GABRE    DBSNP   \n",
       "346533    1.0   0.08              1           0    8907        AP1M1      VEP   \n",
       "346534    1.0   0.08              1           0    8907        AP1M1    DBSNP   \n",
       "346535    NaN    NaN              1           1   10608         MXD4      VEP   \n",
       "346536    NaN    NaN              1           1   10608         MXD4    DBSNP   \n",
       "\n",
       "       geneSymbol_y  DSI_y  DPI_y       PLI       protein_class_name  \\\n",
       "0            INPP4B  0.573  0.577  0.552180                      NaN   \n",
       "1            INPP4B  0.573  0.577  0.552180                      NaN   \n",
       "2            INPP4B  0.573  0.577  0.552180                      NaN   \n",
       "3            INPP4B  0.573  0.577  0.552180                      NaN   \n",
       "4            INPP4B  0.573  0.577  0.552180                      NaN   \n",
       "...             ...    ...    ...       ...                      ...   \n",
       "346532        GABRE  0.839  0.231  0.000184              Ion channel   \n",
       "346533        AP1M1  0.839  0.154  0.712510  Extracellular structure   \n",
       "346534        AP1M1  0.839  0.154  0.712510  Extracellular structure   \n",
       "346535         MXD4  0.722  0.115  0.776350     Transcription factor   \n",
       "346536         MXD4  0.722  0.115  0.776350     Transcription factor   \n",
       "\n",
       "       protein_class  NofDiseases_y  NofPmids_y  \n",
       "0                NaN             90          72  \n",
       "1                NaN             90          72  \n",
       "2                NaN             90          72  \n",
       "3                NaN             90          72  \n",
       "4                NaN             90          72  \n",
       "...              ...            ...         ...  \n",
       "346532  DTO_01300327              8           4  \n",
       "346533  DTO_05007621              5           1  \n",
       "346534  DTO_05007621              5           1  \n",
       "346535  DTO_05007542             19           9  \n",
       "346536  DTO_05007542             19           9  \n",
       "\n",
       "[346537 rows x 20 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#rename the columns for merging\n",
    "\n",
    "Disgenet_all_merged.rename(columns={'geneId': 'GeneID','geneSymbol':'Gene/Protein Symbol'}, inplace=True)\n",
    "Disgenet_all_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "aee58da3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snpId</th>\n",
       "      <th>class</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>DSI_x</th>\n",
       "      <th>DPI_x</th>\n",
       "      <th>NofDiseases_x</th>\n",
       "      <th>NofPmids_x</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>...</th>\n",
       "      <th>DSI_y</th>\n",
       "      <th>DPI_y</th>\n",
       "      <th>PLI</th>\n",
       "      <th>protein_class_name</th>\n",
       "      <th>protein_class</th>\n",
       "      <th>NofDiseases_y</th>\n",
       "      <th>NofPmids_y</th>\n",
       "      <th>Disease Class</th>\n",
       "      <th>Category</th>\n",
       "      <th>Gene/Protein Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3508</td>\n",
       "      <td>...</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.615</td>\n",
       "      <td>3.021200e-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102</td>\n",
       "      <td>60</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3508</td>\n",
       "      <td>...</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.615</td>\n",
       "      <td>3.021200e-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102</td>\n",
       "      <td>60</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1057518588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68908632</td>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>0.851</td>\n",
       "      <td>0.16</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>3508</td>\n",
       "      <td>...</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.615</td>\n",
       "      <td>3.021200e-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102</td>\n",
       "      <td>60</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1057518588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68908632</td>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>0.851</td>\n",
       "      <td>0.16</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>3508</td>\n",
       "      <td>...</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.615</td>\n",
       "      <td>3.021200e-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102</td>\n",
       "      <td>60</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1057518943</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68906163</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>3508</td>\n",
       "      <td>...</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.615</td>\n",
       "      <td>3.021200e-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102</td>\n",
       "      <td>60</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087375</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9358</td>\n",
       "      <td>...</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.423</td>\n",
       "      <td>1.767100e-09</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>27</td>\n",
       "      <td>13</td>\n",
       "      <td>socioeconomic factors</td>\n",
       "      <td>other</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087376</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9358</td>\n",
       "      <td>...</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.423</td>\n",
       "      <td>1.767100e-09</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>27</td>\n",
       "      <td>13</td>\n",
       "      <td>natriuretic peptide, brain</td>\n",
       "      <td>neurological</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087377</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9358</td>\n",
       "      <td>...</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.423</td>\n",
       "      <td>1.767100e-09</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>27</td>\n",
       "      <td>13</td>\n",
       "      <td>waist-hip ratio</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087378</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9358</td>\n",
       "      <td>...</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.423</td>\n",
       "      <td>1.767100e-09</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>27</td>\n",
       "      <td>13</td>\n",
       "      <td>erythrocyte indices</td>\n",
       "      <td>hematological</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087379</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9358</td>\n",
       "      <td>...</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.423</td>\n",
       "      <td>1.767100e-09</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>27</td>\n",
       "      <td>13</td>\n",
       "      <td>socioeconomic factors</td>\n",
       "      <td>other</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2087380 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                snpId class chromosome   position most_severe_consequence  \\\n",
       "0        rs1000091588   snv         11   68935374             stop gained   \n",
       "1        rs1000091588   snv         11   68935374             stop gained   \n",
       "2        rs1057518588   snv         11   68908632    splice donor variant   \n",
       "3        rs1057518588   snv         11   68908632    splice donor variant   \n",
       "4        rs1057518943   snv         11   68906163        missense variant   \n",
       "...               ...   ...        ...        ...                     ...   \n",
       "2087375     rs9557704   snv         13  101575144          intron variant   \n",
       "2087376     rs9557704   snv         13  101575144          intron variant   \n",
       "2087377     rs9557704   snv         13  101575144          intron variant   \n",
       "2087378     rs9557704   snv         13  101575144          intron variant   \n",
       "2087379     rs9557704   snv         13  101575144          intron variant   \n",
       "\n",
       "         DSI_x  DPI_x  NofDiseases_x  NofPmids_x  GeneID  ...  DSI_y  DPI_y  \\\n",
       "0        1.000   0.12              1           1    3508  ...  0.636  0.615   \n",
       "1        1.000   0.12              1           1    3508  ...  0.636  0.615   \n",
       "2        0.851   0.16              6           3    3508  ...  0.636  0.615   \n",
       "3        0.851   0.16              6           3    3508  ...  0.636  0.615   \n",
       "4        1.000   0.12              5           0    3508  ...  0.636  0.615   \n",
       "...        ...    ...            ...         ...     ...  ...    ...    ...   \n",
       "2087375    NaN    NaN              1           1    9358  ...  0.705  0.423   \n",
       "2087376    NaN    NaN              1           1    9358  ...  0.705  0.423   \n",
       "2087377    NaN    NaN              1           1    9358  ...  0.705  0.423   \n",
       "2087378    NaN    NaN              1           1    9358  ...  0.705  0.423   \n",
       "2087379    NaN    NaN              1           1    9358  ...  0.705  0.423   \n",
       "\n",
       "                  PLI  protein_class_name  protein_class  NofDiseases_y  \\\n",
       "0        3.021200e-16                 NaN            NaN            102   \n",
       "1        3.021200e-16                 NaN            NaN            102   \n",
       "2        3.021200e-16                 NaN            NaN            102   \n",
       "3        3.021200e-16                 NaN            NaN            102   \n",
       "4        3.021200e-16                 NaN            NaN            102   \n",
       "...               ...                 ...            ...            ...   \n",
       "2087375  1.767100e-09            Receptor   DTO_05007575             27   \n",
       "2087376  1.767100e-09            Receptor   DTO_05007575             27   \n",
       "2087377  1.767100e-09            Receptor   DTO_05007575             27   \n",
       "2087378  1.767100e-09            Receptor   DTO_05007575             27   \n",
       "2087379  1.767100e-09            Receptor   DTO_05007575             27   \n",
       "\n",
       "        NofPmids_y               Disease Class       Category  \\\n",
       "0               60             iga nephropathy         immune   \n",
       "1               60             iga nephropathy         immune   \n",
       "2               60             iga nephropathy         immune   \n",
       "3               60             iga nephropathy         immune   \n",
       "4               60             iga nephropathy         immune   \n",
       "...            ...                         ...            ...   \n",
       "2087375         13       socioeconomic factors          other   \n",
       "2087376         13  natriuretic peptide, brain   neurological   \n",
       "2087377         13             waist-hip ratio      metabolic   \n",
       "2087378         13         erythrocyte indices  hematological   \n",
       "2087379         13       socioeconomic factors          other   \n",
       "\n",
       "         Gene/Protein Symbol  \n",
       "0                    IGHMBP2  \n",
       "1                    IGHMBP2  \n",
       "2                    IGHMBP2  \n",
       "3                    IGHMBP2  \n",
       "4                    IGHMBP2  \n",
       "...                      ...  \n",
       "2087375               ITGBL1  \n",
       "2087376               ITGBL1  \n",
       "2087377               ITGBL1  \n",
       "2087378               ITGBL1  \n",
       "2087379               ITGBL1  \n",
       "\n",
       "[2087380 rows x 23 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#merging Disgenet + Harmonizome\n",
    "\n",
    "Disgenet_Harmonizome = pd.merge(Disgenet_all_merged, df_Harmonizome, how='inner', on='GeneID')\n",
    "\n",
    "Disgenet_Harmonizome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "9fbcf411",
   "metadata": {},
   "outputs": [],
   "source": [
    "#creating a .csv file\n",
    "Disgenet_Harmonizome.to_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Final\\Harmonizome+Disgenet.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b289479a",
   "metadata": {},
   "source": [
    "# DisGeNET+Harmonizome "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "5c66f803",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\teka_\\AppData\\Local\\Temp\\ipykernel_20056\\2091845720.py:1: DtypeWarning: Columns (2,16,17) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Final\\Harmonizome+Disgenet.csv')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snpId</th>\n",
       "      <th>class</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>DSI_x</th>\n",
       "      <th>DPI_x</th>\n",
       "      <th>NofDiseases_x</th>\n",
       "      <th>NofPmids_x</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>...</th>\n",
       "      <th>DSI_y</th>\n",
       "      <th>DPI_y</th>\n",
       "      <th>PLI</th>\n",
       "      <th>protein_class_name</th>\n",
       "      <th>protein_class</th>\n",
       "      <th>NofDiseases_y</th>\n",
       "      <th>NofPmids_y</th>\n",
       "      <th>Disease Class</th>\n",
       "      <th>Category</th>\n",
       "      <th>Gene/Protein Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3508</td>\n",
       "      <td>...</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.615</td>\n",
       "      <td>3.021200e-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102</td>\n",
       "      <td>60</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3508</td>\n",
       "      <td>...</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.615</td>\n",
       "      <td>3.021200e-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102</td>\n",
       "      <td>60</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1057518588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68908632</td>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>0.851</td>\n",
       "      <td>0.16</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>3508</td>\n",
       "      <td>...</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.615</td>\n",
       "      <td>3.021200e-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102</td>\n",
       "      <td>60</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1057518588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68908632</td>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>0.851</td>\n",
       "      <td>0.16</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>3508</td>\n",
       "      <td>...</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.615</td>\n",
       "      <td>3.021200e-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102</td>\n",
       "      <td>60</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1057518943</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68906163</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>3508</td>\n",
       "      <td>...</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.615</td>\n",
       "      <td>3.021200e-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102</td>\n",
       "      <td>60</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087375</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9358</td>\n",
       "      <td>...</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.423</td>\n",
       "      <td>1.767100e-09</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>27</td>\n",
       "      <td>13</td>\n",
       "      <td>socioeconomic factors</td>\n",
       "      <td>other</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087376</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9358</td>\n",
       "      <td>...</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.423</td>\n",
       "      <td>1.767100e-09</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>27</td>\n",
       "      <td>13</td>\n",
       "      <td>natriuretic peptide, brain</td>\n",
       "      <td>neurological</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087377</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9358</td>\n",
       "      <td>...</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.423</td>\n",
       "      <td>1.767100e-09</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>27</td>\n",
       "      <td>13</td>\n",
       "      <td>waist-hip ratio</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087378</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9358</td>\n",
       "      <td>...</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.423</td>\n",
       "      <td>1.767100e-09</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>27</td>\n",
       "      <td>13</td>\n",
       "      <td>erythrocyte indices</td>\n",
       "      <td>hematological</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087379</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9358</td>\n",
       "      <td>...</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.423</td>\n",
       "      <td>1.767100e-09</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>27</td>\n",
       "      <td>13</td>\n",
       "      <td>socioeconomic factors</td>\n",
       "      <td>other</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2087380 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                snpId class chromosome   position most_severe_consequence  \\\n",
       "0        rs1000091588   snv         11   68935374             stop gained   \n",
       "1        rs1000091588   snv         11   68935374             stop gained   \n",
       "2        rs1057518588   snv         11   68908632    splice donor variant   \n",
       "3        rs1057518588   snv         11   68908632    splice donor variant   \n",
       "4        rs1057518943   snv         11   68906163        missense variant   \n",
       "...               ...   ...        ...        ...                     ...   \n",
       "2087375     rs9557704   snv         13  101575144          intron variant   \n",
       "2087376     rs9557704   snv         13  101575144          intron variant   \n",
       "2087377     rs9557704   snv         13  101575144          intron variant   \n",
       "2087378     rs9557704   snv         13  101575144          intron variant   \n",
       "2087379     rs9557704   snv         13  101575144          intron variant   \n",
       "\n",
       "         DSI_x  DPI_x  NofDiseases_x  NofPmids_x  GeneID  ...  DSI_y  DPI_y  \\\n",
       "0        1.000   0.12              1           1    3508  ...  0.636  0.615   \n",
       "1        1.000   0.12              1           1    3508  ...  0.636  0.615   \n",
       "2        0.851   0.16              6           3    3508  ...  0.636  0.615   \n",
       "3        0.851   0.16              6           3    3508  ...  0.636  0.615   \n",
       "4        1.000   0.12              5           0    3508  ...  0.636  0.615   \n",
       "...        ...    ...            ...         ...     ...  ...    ...    ...   \n",
       "2087375    NaN    NaN              1           1    9358  ...  0.705  0.423   \n",
       "2087376    NaN    NaN              1           1    9358  ...  0.705  0.423   \n",
       "2087377    NaN    NaN              1           1    9358  ...  0.705  0.423   \n",
       "2087378    NaN    NaN              1           1    9358  ...  0.705  0.423   \n",
       "2087379    NaN    NaN              1           1    9358  ...  0.705  0.423   \n",
       "\n",
       "                  PLI  protein_class_name  protein_class  NofDiseases_y  \\\n",
       "0        3.021200e-16                 NaN            NaN            102   \n",
       "1        3.021200e-16                 NaN            NaN            102   \n",
       "2        3.021200e-16                 NaN            NaN            102   \n",
       "3        3.021200e-16                 NaN            NaN            102   \n",
       "4        3.021200e-16                 NaN            NaN            102   \n",
       "...               ...                 ...            ...            ...   \n",
       "2087375  1.767100e-09            Receptor   DTO_05007575             27   \n",
       "2087376  1.767100e-09            Receptor   DTO_05007575             27   \n",
       "2087377  1.767100e-09            Receptor   DTO_05007575             27   \n",
       "2087378  1.767100e-09            Receptor   DTO_05007575             27   \n",
       "2087379  1.767100e-09            Receptor   DTO_05007575             27   \n",
       "\n",
       "        NofPmids_y               Disease Class       Category  \\\n",
       "0               60             iga nephropathy         immune   \n",
       "1               60             iga nephropathy         immune   \n",
       "2               60             iga nephropathy         immune   \n",
       "3               60             iga nephropathy         immune   \n",
       "4               60             iga nephropathy         immune   \n",
       "...            ...                         ...            ...   \n",
       "2087375         13       socioeconomic factors          other   \n",
       "2087376         13  natriuretic peptide, brain   neurological   \n",
       "2087377         13             waist-hip ratio      metabolic   \n",
       "2087378         13         erythrocyte indices  hematological   \n",
       "2087379         13       socioeconomic factors          other   \n",
       "\n",
       "         Gene/Protein Symbol  \n",
       "0                    IGHMBP2  \n",
       "1                    IGHMBP2  \n",
       "2                    IGHMBP2  \n",
       "3                    IGHMBP2  \n",
       "4                    IGHMBP2  \n",
       "...                      ...  \n",
       "2087375               ITGBL1  \n",
       "2087376               ITGBL1  \n",
       "2087377               ITGBL1  \n",
       "2087378               ITGBL1  \n",
       "2087379               ITGBL1  \n",
       "\n",
       "[2087380 rows x 23 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Loading the CSV file into a DataFrame\n",
    "df = pd.read_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Final\\Harmonizome+Disgenet.csv')\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4520245f",
   "metadata": {},
   "source": [
    "The genes most commonly associated with breast cancer include:\n",
    "\n",
    "BRCA1 (Breast Cancer 1) Function: Tumor suppressor gene that helps repair damaged DNA. Association: Mutations in BRCA1 significantly increase the risk of developing breast cancer and ovarian cancer.\n",
    "\n",
    "BRCA2 (Breast Cancer 2) Function: Similar to BRCA1, BRCA2 is also a tumor suppressor gene involved in DNA repair. Association: Mutations in BRCA2 increase the risk of breast cancer, ovarian cancer, and other types of cancer.\n",
    "\n",
    "TP53 (Tumor Protein p53) Function: Tumor suppressor gene that regulates the cell cycle and prevents the proliferation of cells with damaged DNA. Association: Mutations in TP53 are found in several types of cancer, including breast cancer.\n",
    "\n",
    "HER2 (Human Epidermal Growth Factor Receptor 2) Function: Gene that encodes a growth factor receptor protein, involved in cell growth. Association: HER2 overexpression is associated with a more aggressive type of breast cancer.\n",
    "\n",
    "PIK3CA (Phosphatidylinositol-4,5-bisphosphate 3-kinase catalytic subunit alpha) Function: Involved in cell growth and survival signals. Association: Mutations in PIK3CA are common in breast cancer and may influence response to treatment.\n",
    "\n",
    "PTEN (Phosphatase and Tensin Homolog) Function: Tumor suppressor gene that regulates cell growth. Association: Mutations in PTEN can lead to uncontrolled cell growth and are associated with several types of cancer, including breast cancer.\n",
    "\n",
    "CDH1 (Cadherin 1) Function: Involved in cell adhesion and maintenance of tissue structure. Association: Mutations in CDH1 are associated with lobular breast cancer.\n",
    "\n",
    "PALB2 (Partner and Localizer of BRCA2) Function: Works together with BRCA2 in DNA repair. Association: Mutations in PALB2 increase the risk of breast cancer.\n",
    "\n",
    "ATM (Ataxia-Telangiectasia Mutated) Function: Involved in the response to DNA damage. Association: Mutations in ATM are associated with an increased risk of breast cancer.\n",
    "\n",
    "CHEK2 (Checkpoint Kinase 2) Function: Acts in the response to DNA damage. Association: Mutations in CHEK2 are associated with a moderate increase in the risk of breast cancer."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "5ca045ee",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene/Protein Symbol\n",
      "BRCA1     280800\n",
      "TP53      176943\n",
      "BRCA2      42924\n",
      "ATM        25298\n",
      "PTEN        3840\n",
      "CHEK2       3132\n",
      "CDH1        2597\n",
      "PALB2       2460\n",
      "PIK3CA      2158\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# List of genes to search for\n",
    "genes_to_find = ['BRCA1', 'BRCA2', 'TP53', 'HER2', 'PIK3CA', 'PTEN', 'CDH1', 'PALB2', 'ATM', 'CHEK2']\n",
    "\n",
    "# Filter the DataFrame to include only rows with the specific genes\n",
    "df_genes_filtered = df[df['Gene/Protein Symbol'].isin(genes_to_find)]\n",
    "\n",
    "# Use value_counts to count occurrences of each gene\n",
    "gene_counts = df_genes_filtered['Gene/Protein Symbol'].value_counts()\n",
    "\n",
    "# Display the counts\n",
    "print(gene_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "4a32b081",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snpId</th>\n",
       "      <th>class</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>sourceId</th>\n",
       "      <th>protein_class_name</th>\n",
       "      <th>protein_class</th>\n",
       "      <th>Disease Class</th>\n",
       "      <th>Category</th>\n",
       "      <th>Gene/Protein Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>3508</td>\n",
       "      <td>VEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>3508</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1057518588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68908632</td>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>3508</td>\n",
       "      <td>VEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1057518588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68908632</td>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>3508</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1057518943</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68906163</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>3508</td>\n",
       "      <td>VEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087375</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>socioeconomic factors</td>\n",
       "      <td>other</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087376</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>natriuretic peptide, brain</td>\n",
       "      <td>neurological</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087377</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>waist-hip ratio</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087378</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>erythrocyte indices</td>\n",
       "      <td>hematological</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087379</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>socioeconomic factors</td>\n",
       "      <td>other</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2087380 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                snpId class chromosome   position most_severe_consequence  \\\n",
       "0        rs1000091588   snv         11   68935374             stop gained   \n",
       "1        rs1000091588   snv         11   68935374             stop gained   \n",
       "2        rs1057518588   snv         11   68908632    splice donor variant   \n",
       "3        rs1057518588   snv         11   68908632    splice donor variant   \n",
       "4        rs1057518943   snv         11   68906163        missense variant   \n",
       "...               ...   ...        ...        ...                     ...   \n",
       "2087375     rs9557704   snv         13  101575144          intron variant   \n",
       "2087376     rs9557704   snv         13  101575144          intron variant   \n",
       "2087377     rs9557704   snv         13  101575144          intron variant   \n",
       "2087378     rs9557704   snv         13  101575144          intron variant   \n",
       "2087379     rs9557704   snv         13  101575144          intron variant   \n",
       "\n",
       "         GeneID sourceId protein_class_name protein_class  \\\n",
       "0          3508      VEP                NaN           NaN   \n",
       "1          3508    DBSNP                NaN           NaN   \n",
       "2          3508      VEP                NaN           NaN   \n",
       "3          3508    DBSNP                NaN           NaN   \n",
       "4          3508      VEP                NaN           NaN   \n",
       "...         ...      ...                ...           ...   \n",
       "2087375    9358      VEP           Receptor  DTO_05007575   \n",
       "2087376    9358    DBSNP           Receptor  DTO_05007575   \n",
       "2087377    9358    DBSNP           Receptor  DTO_05007575   \n",
       "2087378    9358    DBSNP           Receptor  DTO_05007575   \n",
       "2087379    9358    DBSNP           Receptor  DTO_05007575   \n",
       "\n",
       "                      Disease Class       Category Gene/Protein Symbol  \n",
       "0                   iga nephropathy         immune             IGHMBP2  \n",
       "1                   iga nephropathy         immune             IGHMBP2  \n",
       "2                   iga nephropathy         immune             IGHMBP2  \n",
       "3                   iga nephropathy         immune             IGHMBP2  \n",
       "4                   iga nephropathy         immune             IGHMBP2  \n",
       "...                             ...            ...                 ...  \n",
       "2087375       socioeconomic factors          other              ITGBL1  \n",
       "2087376  natriuretic peptide, brain   neurological              ITGBL1  \n",
       "2087377             waist-hip ratio      metabolic              ITGBL1  \n",
       "2087378         erythrocyte indices  hematological              ITGBL1  \n",
       "2087379       socioeconomic factors          other              ITGBL1  \n",
       "\n",
       "[2087380 rows x 12 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#dropping duplicated columns\n",
    "\n",
    "df = df.drop(columns=['DSI_y', 'DPI_y','NofDiseases_y', 'NofPmids_y', 'geneSymbol_y', 'geneSymbol_x', 'DSI_x', 'DPI_x', 'NofDiseases_x', 'NofPmids_x', 'PLI']).reset_index(drop=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "8a9f09d7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snpID</th>\n",
       "      <th>class</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>sourceId</th>\n",
       "      <th>protein_class_name</th>\n",
       "      <th>protein_class</th>\n",
       "      <th>Disease Class</th>\n",
       "      <th>Category</th>\n",
       "      <th>Gene/Protein Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>3508</td>\n",
       "      <td>VEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>3508</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1057518588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68908632</td>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>3508</td>\n",
       "      <td>VEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1057518588</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68908632</td>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>3508</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1057518943</td>\n",
       "      <td>snv</td>\n",
       "      <td>11</td>\n",
       "      <td>68906163</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>3508</td>\n",
       "      <td>VEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>iga nephropathy</td>\n",
       "      <td>immune</td>\n",
       "      <td>IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087375</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>socioeconomic factors</td>\n",
       "      <td>other</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087376</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>natriuretic peptide, brain</td>\n",
       "      <td>neurological</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087377</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>waist-hip ratio</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087378</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>erythrocyte indices</td>\n",
       "      <td>hematological</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087379</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>socioeconomic factors</td>\n",
       "      <td>other</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2087380 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                snpID class chromosome   position most_severe_consequence  \\\n",
       "0        rs1000091588   snv         11   68935374             stop gained   \n",
       "1        rs1000091588   snv         11   68935374             stop gained   \n",
       "2        rs1057518588   snv         11   68908632    splice donor variant   \n",
       "3        rs1057518588   snv         11   68908632    splice donor variant   \n",
       "4        rs1057518943   snv         11   68906163        missense variant   \n",
       "...               ...   ...        ...        ...                     ...   \n",
       "2087375     rs9557704   snv         13  101575144          intron variant   \n",
       "2087376     rs9557704   snv         13  101575144          intron variant   \n",
       "2087377     rs9557704   snv         13  101575144          intron variant   \n",
       "2087378     rs9557704   snv         13  101575144          intron variant   \n",
       "2087379     rs9557704   snv         13  101575144          intron variant   \n",
       "\n",
       "         GeneID sourceId protein_class_name protein_class  \\\n",
       "0          3508      VEP                NaN           NaN   \n",
       "1          3508    DBSNP                NaN           NaN   \n",
       "2          3508      VEP                NaN           NaN   \n",
       "3          3508    DBSNP                NaN           NaN   \n",
       "4          3508      VEP                NaN           NaN   \n",
       "...         ...      ...                ...           ...   \n",
       "2087375    9358      VEP           Receptor  DTO_05007575   \n",
       "2087376    9358    DBSNP           Receptor  DTO_05007575   \n",
       "2087377    9358    DBSNP           Receptor  DTO_05007575   \n",
       "2087378    9358    DBSNP           Receptor  DTO_05007575   \n",
       "2087379    9358    DBSNP           Receptor  DTO_05007575   \n",
       "\n",
       "                      Disease Class       Category Gene/Protein Symbol  \n",
       "0                   iga nephropathy         immune             IGHMBP2  \n",
       "1                   iga nephropathy         immune             IGHMBP2  \n",
       "2                   iga nephropathy         immune             IGHMBP2  \n",
       "3                   iga nephropathy         immune             IGHMBP2  \n",
       "4                   iga nephropathy         immune             IGHMBP2  \n",
       "...                             ...            ...                 ...  \n",
       "2087375       socioeconomic factors          other              ITGBL1  \n",
       "2087376  natriuretic peptide, brain   neurological              ITGBL1  \n",
       "2087377             waist-hip ratio      metabolic              ITGBL1  \n",
       "2087378         erythrocyte indices  hematological              ITGBL1  \n",
       "2087379       socioeconomic factors          other              ITGBL1  \n",
       "\n",
       "[2087380 rows x 12 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##rename the columns post-merging\n",
    "\n",
    "df.rename(columns={'snpId': 'snpID'}, inplace=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "b67c7045",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2087380, 12)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "f789cc88",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 2087380 entries, 0 to 2087379\n",
      "Data columns (total 12 columns):\n",
      " #   Column                   Dtype \n",
      "---  ------                   ----- \n",
      " 0   snpID                    object\n",
      " 1   class                    object\n",
      " 2   chromosome               object\n",
      " 3   position                 int64 \n",
      " 4   most_severe_consequence  object\n",
      " 5   GeneID                   int64 \n",
      " 6   sourceId                 object\n",
      " 7   protein_class_name       object\n",
      " 8   protein_class            object\n",
      " 9   Disease Class            object\n",
      " 10  Category                 object\n",
      " 11  Gene/Protein Symbol      object\n",
      "dtypes: int64(2), object(10)\n",
      "memory usage: 191.1+ MB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "89cf4444",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "snpID                      object\n",
       "class                      object\n",
       "chromosome                 object\n",
       "position                    int64\n",
       "most_severe_consequence    object\n",
       "GeneID                      int64\n",
       "sourceId                   object\n",
       "protein_class_name         object\n",
       "protein_class              object\n",
       "Disease Class              object\n",
       "Category                   object\n",
       "Gene/Protein Symbol        object\n",
       "dtype: object"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "427ec57d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>GeneID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>2.087380e+06</td>\n",
       "      <td>2.087380e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>5.912527e+07</td>\n",
       "      <td>8.002058e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>5.045177e+07</td>\n",
       "      <td>2.722170e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>9.610000e+02</td>\n",
       "      <td>2.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>2.272703e+07</td>\n",
       "      <td>6.750000e+02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>4.309372e+07</td>\n",
       "      <td>3.043000e+03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>8.893295e+07</td>\n",
       "      <td>5.053000e+03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>2.474493e+08</td>\n",
       "      <td>1.005074e+08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           position        GeneID\n",
       "count  2.087380e+06  2.087380e+06\n",
       "mean   5.912527e+07  8.002058e+04\n",
       "std    5.045177e+07  2.722170e+06\n",
       "min    9.610000e+02  2.000000e+00\n",
       "25%    2.272703e+07  6.750000e+02\n",
       "50%    4.309372e+07  3.043000e+03\n",
       "75%    8.893295e+07  5.053000e+03\n",
       "max    2.474493e+08  1.005074e+08"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "73225d0c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "snpID                           0\n",
       "class                           0\n",
       "chromosome                      0\n",
       "position                        0\n",
       "most_severe_consequence         0\n",
       "GeneID                          0\n",
       "sourceId                        0\n",
       "protein_class_name         596419\n",
       "protein_class              596419\n",
       "Disease Class                   0\n",
       "Category                        0\n",
       "Gene/Protein Symbol             0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nan_per_column = df.isna().sum()\n",
    "nan_per_column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "105e1ab9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snpID</th>\n",
       "      <th>class</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>sourceId</th>\n",
       "      <th>protein_class_name</th>\n",
       "      <th>protein_class</th>\n",
       "      <th>Disease Class</th>\n",
       "      <th>Category</th>\n",
       "      <th>Gene/Protein Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>rs10001410</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>61608511</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>23284</td>\n",
       "      <td>VEP</td>\n",
       "      <td>G-protein coupled receptor</td>\n",
       "      <td>DTO_02300001</td>\n",
       "      <td>biochemical measures</td>\n",
       "      <td>other</td>\n",
       "      <td>LPHN3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>rs10001410</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>61608511</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>23284</td>\n",
       "      <td>VEP</td>\n",
       "      <td>G-protein coupled receptor</td>\n",
       "      <td>DTO_02300001</td>\n",
       "      <td>memory, short-term</td>\n",
       "      <td>aging</td>\n",
       "      <td>LPHN3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>rs10001410</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>61608511</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>23284</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>G-protein coupled receptor</td>\n",
       "      <td>DTO_02300001</td>\n",
       "      <td>biochemical measures</td>\n",
       "      <td>other</td>\n",
       "      <td>LPHN3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>rs10001410</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>61608511</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>23284</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>G-protein coupled receptor</td>\n",
       "      <td>DTO_02300001</td>\n",
       "      <td>memory, short-term</td>\n",
       "      <td>aging</td>\n",
       "      <td>LPHN3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>rs1355368</td>\n",
       "      <td>snv</td>\n",
       "      <td>4</td>\n",
       "      <td>61914104</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>23284</td>\n",
       "      <td>VEP</td>\n",
       "      <td>G-protein coupled receptor</td>\n",
       "      <td>DTO_02300001</td>\n",
       "      <td>biochemical measures</td>\n",
       "      <td>other</td>\n",
       "      <td>LPHN3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087375</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>socioeconomic factors</td>\n",
       "      <td>other</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087376</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>natriuretic peptide, brain</td>\n",
       "      <td>neurological</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087377</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>waist-hip ratio</td>\n",
       "      <td>metabolic</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087378</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>erythrocyte indices</td>\n",
       "      <td>hematological</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087379</th>\n",
       "      <td>rs9557704</td>\n",
       "      <td>snv</td>\n",
       "      <td>13</td>\n",
       "      <td>101575144</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>9358</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Receptor</td>\n",
       "      <td>DTO_05007575</td>\n",
       "      <td>socioeconomic factors</td>\n",
       "      <td>other</td>\n",
       "      <td>ITGBL1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1490961 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              snpID class chromosome   position most_severe_consequence  \\\n",
       "174      rs10001410   snv          4   61608511          intron variant   \n",
       "175      rs10001410   snv          4   61608511          intron variant   \n",
       "176      rs10001410   snv          4   61608511          intron variant   \n",
       "177      rs10001410   snv          4   61608511          intron variant   \n",
       "178       rs1355368   snv          4   61914104          intron variant   \n",
       "...             ...   ...        ...        ...                     ...   \n",
       "2087375   rs9557704   snv         13  101575144          intron variant   \n",
       "2087376   rs9557704   snv         13  101575144          intron variant   \n",
       "2087377   rs9557704   snv         13  101575144          intron variant   \n",
       "2087378   rs9557704   snv         13  101575144          intron variant   \n",
       "2087379   rs9557704   snv         13  101575144          intron variant   \n",
       "\n",
       "         GeneID sourceId          protein_class_name protein_class  \\\n",
       "174       23284      VEP  G-protein coupled receptor  DTO_02300001   \n",
       "175       23284      VEP  G-protein coupled receptor  DTO_02300001   \n",
       "176       23284    DBSNP  G-protein coupled receptor  DTO_02300001   \n",
       "177       23284    DBSNP  G-protein coupled receptor  DTO_02300001   \n",
       "178       23284      VEP  G-protein coupled receptor  DTO_02300001   \n",
       "...         ...      ...                         ...           ...   \n",
       "2087375    9358      VEP                    Receptor  DTO_05007575   \n",
       "2087376    9358    DBSNP                    Receptor  DTO_05007575   \n",
       "2087377    9358    DBSNP                    Receptor  DTO_05007575   \n",
       "2087378    9358    DBSNP                    Receptor  DTO_05007575   \n",
       "2087379    9358    DBSNP                    Receptor  DTO_05007575   \n",
       "\n",
       "                      Disease Class       Category Gene/Protein Symbol  \n",
       "174            biochemical measures          other               LPHN3  \n",
       "175              memory, short-term          aging               LPHN3  \n",
       "176            biochemical measures          other               LPHN3  \n",
       "177              memory, short-term          aging               LPHN3  \n",
       "178            biochemical measures          other               LPHN3  \n",
       "...                             ...            ...                 ...  \n",
       "2087375       socioeconomic factors          other              ITGBL1  \n",
       "2087376  natriuretic peptide, brain   neurological              ITGBL1  \n",
       "2087377             waist-hip ratio      metabolic              ITGBL1  \n",
       "2087378         erythrocyte indices  hematological              ITGBL1  \n",
       "2087379       socioeconomic factors          other              ITGBL1  \n",
       "\n",
       "[1490961 rows x 12 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.dropna(inplace=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "9ae0d15b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "snpID                      0\n",
       "class                      0\n",
       "chromosome                 0\n",
       "position                   0\n",
       "most_severe_consequence    0\n",
       "GeneID                     0\n",
       "sourceId                   0\n",
       "protein_class_name         0\n",
       "protein_class              0\n",
       "Disease Class              0\n",
       "Category                   0\n",
       "Gene/Protein Symbol        0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nan_per_column = df.isna().sum()\n",
    "nan_per_column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "78e5593c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['snpID', 'class', 'chromosome', 'position', 'most_severe_consequence',\n",
       "       'GeneID', 'sourceId', 'protein_class_name', 'protein_class',\n",
       "       'Disease Class', 'Category', 'Gene/Protein Symbol'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "7c0af9a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene/Protein Symbol\n",
       "BRCA1      280800\n",
       "TP53       176943\n",
       "CFTR       100125\n",
       "ACE         64480\n",
       "MLH1        48448\n",
       "            ...  \n",
       "CYP26A1         1\n",
       "TIMP4           1\n",
       "TAS1R3          1\n",
       "SLC22A6         1\n",
       "TFF2            1\n",
       "Name: count, Length: 1333, dtype: int64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Gene/Protein Symbol'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6eda9de4",
   "metadata": {},
   "source": [
    "The genes most commonly associated with breast cancer include:\n",
    "\n",
    "BRCA1 (Breast Cancer 1)\n",
    "Function: Tumor suppressor gene that helps repair damaged DNA.\n",
    "Association: Mutations in BRCA1 significantly increase the risk of developing breast cancer and ovarian cancer.\n",
    "\n",
    "BRCA2 (Breast Cancer 2)\n",
    "Function: Similar to BRCA1, BRCA2 is also a tumor suppressor gene involved in DNA repair.\n",
    "Association: Mutations in BRCA2 increase the risk of breast cancer, ovarian cancer, and other types of cancer.\n",
    "\n",
    "TP53 (Tumor Protein p53)\n",
    "Function: Tumor suppressor gene that regulates the cell cycle and prevents the proliferation of cells with damaged DNA.\n",
    "Association: Mutations in TP53 are found in several types of cancer, including breast cancer.\n",
    "\n",
    "HER2 (Human Epidermal Growth Factor Receptor 2)\n",
    "Function: Gene that encodes a growth factor receptor protein, involved in cell growth.\n",
    "Association: HER2 overexpression is associated with a more aggressive type of breast cancer.\n",
    "\n",
    "PIK3CA (Phosphatidylinositol-4,5-bisphosphate 3-kinase catalytic subunit alpha)\n",
    "Function: Involved in cell growth and survival signals.\n",
    "Association: Mutations in PIK3CA are common in breast cancer and may influence response to treatment.\n",
    "\n",
    "PTEN (Phosphatase and Tensin Homolog)\n",
    "Function: Tumor suppressor gene that regulates cell growth.\n",
    "Association: Mutations in PTEN can lead to uncontrolled cell growth and are associated with several types of cancer, including breast cancer.\n",
    "\n",
    "CDH1 (Cadherin 1)\n",
    "Function: Involved in cell adhesion and maintenance of tissue structure.\n",
    "Association: Mutations in CDH1 are associated with lobular breast cancer.\n",
    "\n",
    "PALB2 (Partner and Localizer of BRCA2)\n",
    "Function: Works together with BRCA2 in DNA repair.\n",
    "Association: Mutations in PALB2 increase the risk of breast cancer.\n",
    "\n",
    "ATM (Ataxia-Telangiectasia Mutated)\n",
    "Function: Involved in the response to DNA damage.\n",
    "Association: Mutations in ATM are associated with an increased risk of breast cancer.\n",
    "\n",
    "CHEK2 (Checkpoint Kinase 2)\n",
    "Function: Acts in the response to DNA damage.\n",
    "Association: Mutations in CHEK2 are associated with a moderate increase in the risk of breast cancer."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "1832df67",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene/Protein Symbol\n",
      "BRCA1     280800\n",
      "TP53      176943\n",
      "BRCA2      42924\n",
      "ATM        25298\n",
      "PTEN        3840\n",
      "CHEK2       3132\n",
      "PIK3CA      2158\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# List of genes to search for\n",
    "genes_to_find = ['BRCA1', 'BRCA2', 'TP53', 'HER2', 'PIK3CA', 'PTEN', 'CDH1', 'PALB2', 'ATM', 'CHEK2']\n",
    "\n",
    "# Filtering the DataFrame to contain only the rows with the specific genes\n",
    "df_genes_filtered = df[df['Gene/Protein Symbol'].isin(genes_to_find)]\n",
    "\n",
    "# Using value_counts to count the occurrences of each gene\n",
    "gene_counts = df_genes_filtered['Gene/Protein Symbol'].value_counts()\n",
    "\n",
    "# Displaying the counts\n",
    "print(gene_counts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d99541ae",
   "metadata": {},
   "source": [
    "#renaming the synomym GeneIDs as per official names\n",
    "#both Harmozinome and DisGENet utilize the \"NCBI Entrez\" for gene identification. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "0118e401",
   "metadata": {},
   "outputs": [],
   "source": [
    "def rename_genes_in_dataframe(df, gene_input):\n",
    "    \"\"\"\n",
    "    Function to rename gene synonyms in a DataFrame.\n",
    "\n",
    "    Parameters:\n",
    "    df (DataFrame): The DataFrame where the genes will be renamed.\n",
    "    gene_input (str): String with gene synonyms and common name in the format 'synonyms; common_name'.\n",
    "\n",
    "    Returns:\n",
    "    DataFrame: The updated DataFrame with genes renamed.\n",
    "    \"\"\"\n",
    "    # Separate the synonyms from the common name\n",
    "    gene_synonyms, common_name = gene_input.split(';')\n",
    "    \n",
    "    # Convert the synonym string into a list\n",
    "    synonym_list = [gene.strip() for gene in gene_synonyms.split(',')]\n",
    "    \n",
    "    # Replace the synonyms with the common name in the DataFrame\n",
    "    df.replace(synonym_list, common_name.strip(), inplace=True)\n",
    "    \n",
    "    return df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "d4ca23de",
   "metadata": {},
   "outputs": [],
   "source": [
    "# List of gene inputs to be renamed\n",
    "gene_inputs = [\n",
    "    \"BCC7, LFS1, P53, TRP53; TP53\",\n",
    "    \"BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNCA4, PPP1R53, PSCP, RNF53; BRCA1\",\n",
    "    \"AT1, ATA, ATC, ATD, ATDC, ATE, TEL1, TELO1; ATM\",\n",
    "    \"BRCC2, BROVCA2, FACD, FAD, FAD1, FANCD, FANCD1, GLM3, PNCA2, XRCC11, RP11-298P3.4; BRCA2\",\n",
    "    \"10q23del, BZS, CWS1, DEC, GLM2, MHAM, MMAC1, PTEN1, TEP1; PTEN\",\n",
    "    \"CDS1, CHK2, HuCds1, LFS2, PP1425, RAD53, hCds1, RP11-436C9.1; CHEK2\",\n",
    "    \"CLOVE, CWS5, MCAP, MCM, MCMTC, PI3K, p110-alpha; PIK3CA\"]\n",
    "\n",
    "# Loop to apply the function to each entry\n",
    "for gene_input in gene_inputs:\n",
    "    df = rename_genes_in_dataframe(df, gene_input)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "7da94983",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snpID</th>\n",
       "      <th>class</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>sourceId</th>\n",
       "      <th>protein_class_name</th>\n",
       "      <th>protein_class</th>\n",
       "      <th>Disease Class</th>\n",
       "      <th>Category</th>\n",
       "      <th>Gene/Protein Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000256867</td>\n",
       "      <td>snv</td>\n",
       "      <td>17</td>\n",
       "      <td>7673550</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7157</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Transcription factor</td>\n",
       "      <td>DTO_05007542</td>\n",
       "      <td>cervical intraepithelial neoplasia; papillomav...</td>\n",
       "      <td>cancer</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1000256867</td>\n",
       "      <td>snv</td>\n",
       "      <td>17</td>\n",
       "      <td>7673550</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7157</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Transcription factor</td>\n",
       "      <td>DTO_05007542</td>\n",
       "      <td>carcinoma, squamous cell; esophageal neoplasms...</td>\n",
       "      <td>cancer</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1000256867</td>\n",
       "      <td>snv</td>\n",
       "      <td>17</td>\n",
       "      <td>7673550</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7157</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Transcription factor</td>\n",
       "      <td>DTO_05007542</td>\n",
       "      <td>duodenal ulcer; helicobacter infections; stoma...</td>\n",
       "      <td>cancer</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1000256867</td>\n",
       "      <td>snv</td>\n",
       "      <td>17</td>\n",
       "      <td>7673550</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7157</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Transcription factor</td>\n",
       "      <td>DTO_05007542</td>\n",
       "      <td>oral lichen planus</td>\n",
       "      <td>infection</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1000256867</td>\n",
       "      <td>snv</td>\n",
       "      <td>17</td>\n",
       "      <td>7673550</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7157</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Transcription factor</td>\n",
       "      <td>DTO_05007542</td>\n",
       "      <td>kaposi's sarcoma</td>\n",
       "      <td>cancer</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>535090</th>\n",
       "      <td>rs967624732</td>\n",
       "      <td>snv</td>\n",
       "      <td>17</td>\n",
       "      <td>43097269</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>672</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Enzyme</td>\n",
       "      <td>DTO_05007624</td>\n",
       "      <td>breast neoplasms; cell transformation, neoplas...</td>\n",
       "      <td>cancer</td>\n",
       "      <td>BRCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>535091</th>\n",
       "      <td>rs967624732</td>\n",
       "      <td>snv</td>\n",
       "      <td>17</td>\n",
       "      <td>43097269</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>672</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Enzyme</td>\n",
       "      <td>DTO_05007624</td>\n",
       "      <td>breast cancer, male</td>\n",
       "      <td>cancer</td>\n",
       "      <td>BRCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>535092</th>\n",
       "      <td>rs967624732</td>\n",
       "      <td>snv</td>\n",
       "      <td>17</td>\n",
       "      <td>43097269</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>672</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Enzyme</td>\n",
       "      <td>DTO_05007624</td>\n",
       "      <td>breast neoplasms; mammary neoplasms; neoplasms...</td>\n",
       "      <td>cancer</td>\n",
       "      <td>BRCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>535093</th>\n",
       "      <td>rs967624732</td>\n",
       "      <td>snv</td>\n",
       "      <td>17</td>\n",
       "      <td>43097269</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>672</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Enzyme</td>\n",
       "      <td>DTO_05007624</td>\n",
       "      <td>ovarian diseases</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>BRCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>535094</th>\n",
       "      <td>rs967624732</td>\n",
       "      <td>snv</td>\n",
       "      <td>17</td>\n",
       "      <td>43097269</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>672</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Enzyme</td>\n",
       "      <td>DTO_05007624</td>\n",
       "      <td>endometrial carcinoma</td>\n",
       "      <td>cancer</td>\n",
       "      <td>BRCA1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>535095 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               snpID class chromosome  position most_severe_consequence  \\\n",
       "0       rs1000256867   snv         17   7673550        missense variant   \n",
       "1       rs1000256867   snv         17   7673550        missense variant   \n",
       "2       rs1000256867   snv         17   7673550        missense variant   \n",
       "3       rs1000256867   snv         17   7673550        missense variant   \n",
       "4       rs1000256867   snv         17   7673550        missense variant   \n",
       "...              ...   ...        ...       ...                     ...   \n",
       "535090   rs967624732   snv         17  43097269        missense variant   \n",
       "535091   rs967624732   snv         17  43097269        missense variant   \n",
       "535092   rs967624732   snv         17  43097269        missense variant   \n",
       "535093   rs967624732   snv         17  43097269        missense variant   \n",
       "535094   rs967624732   snv         17  43097269        missense variant   \n",
       "\n",
       "        GeneID sourceId    protein_class_name protein_class  \\\n",
       "0         7157      VEP  Transcription factor  DTO_05007542   \n",
       "1         7157      VEP  Transcription factor  DTO_05007542   \n",
       "2         7157      VEP  Transcription factor  DTO_05007542   \n",
       "3         7157      VEP  Transcription factor  DTO_05007542   \n",
       "4         7157      VEP  Transcription factor  DTO_05007542   \n",
       "...        ...      ...                   ...           ...   \n",
       "535090     672      VEP                Enzyme  DTO_05007624   \n",
       "535091     672      VEP                Enzyme  DTO_05007624   \n",
       "535092     672      VEP                Enzyme  DTO_05007624   \n",
       "535093     672      VEP                Enzyme  DTO_05007624   \n",
       "535094     672      VEP                Enzyme  DTO_05007624   \n",
       "\n",
       "                                            Disease Class      Category  \\\n",
       "0       cervical intraepithelial neoplasia; papillomav...        cancer   \n",
       "1       carcinoma, squamous cell; esophageal neoplasms...        cancer   \n",
       "2       duodenal ulcer; helicobacter infections; stoma...        cancer   \n",
       "3                                      oral lichen planus     infection   \n",
       "4                                        kaposi's sarcoma        cancer   \n",
       "...                                                   ...           ...   \n",
       "535090  breast neoplasms; cell transformation, neoplas...        cancer   \n",
       "535091                                breast cancer, male        cancer   \n",
       "535092  breast neoplasms; mammary neoplasms; neoplasms...        cancer   \n",
       "535093                                   ovarian diseases  reproduction   \n",
       "535094                              endometrial carcinoma        cancer   \n",
       "\n",
       "       Gene/Protein Symbol  \n",
       "0                     TP53  \n",
       "1                     TP53  \n",
       "2                     TP53  \n",
       "3                     TP53  \n",
       "4                     TP53  \n",
       "...                    ...  \n",
       "535090               BRCA1  \n",
       "535091               BRCA1  \n",
       "535092               BRCA1  \n",
       "535093               BRCA1  \n",
       "535094               BRCA1  \n",
       "\n",
       "[535095 rows x 12 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# List of genes to keep\n",
    "genes_to_keep = ['TP53', 'BRCA1','ATM', 'BRCA2', 'PTEN', 'CHEK2', 'PIK3CA']\n",
    "\n",
    "# Filtering the DataFrame to keep only the specific genes\n",
    "df = df[df['Gene/Protein Symbol'].isin(genes_to_keep)].reset_index(drop=True)\n",
    "\n",
    "# Displaying the updated DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "9dae3969",
   "metadata": {},
   "outputs": [],
   "source": [
    "#CREATING A CLEAN DATA \n",
    "df.to_csv(r'C:\\Users\\teka_\\DATA SCIENCE\\Concordia-Bootcamps\\FINAL PROJECT\\Jupyter Notebook\\1-used\\df_wo_othersGens.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
